### PR TITLE
add l2_loss function to tensorflow frontend

### PR DESCRIPTION
**Pull request by 0StevenH0**
# PR Description 
- Added l2_loss function at ivy/experimental/nn.py
- Added l2_loss function at tensorflow frontend
- Added l2_loss test

## Related 

Close #23073

## Checklist 

- [x] Did you add a function?
- [x] Did you add the tests?
- [x] Did you follow the steps we provided?

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_nn.py [hunk 1]
```diff
@@ -1397,6 +1397,41 @@ def test_tensorflow_log_poisson_loss(
...
 
+@handle_frontend_test(
+    fn_tree="tensorflow.nn.l2_loss",
+    dtype_target_inputs=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"),
+        num_arrays=1,
+        min_value=0,
+        max_value=1,
+        min_num_dims=1,
+        max_num_dims=5,
```
Comment (KareemMAX) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_nn.py:1408]: Are those necessary?

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_nn.py [hunk 2]
```diff
@@ -1397,6 +1397,41 @@ def test_tensorflow_log_poisson_loss(
...
+        shared_dtype=True,
+    ),
+    compute_full_loss=st.booleans(),
```
Comment (KareemMAX) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_nn.py:1411]: This term is not used

**Review by KareemMAX — CHANGES_REQUESTED** [review]
Hey @0StevenH0,
Great work! I guess you are still working on adding the docstrings. Also I guess we would need a functional test for this function.

I also pointed out some comment on the tests.

Thanks 🚀

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_nn.py (hunk 2 above)
Comment (0StevenH0) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_nn.py:1406]: hey @KareemMAX thanks for the review, i tried do as you said, and as you said, these arguments except for compute_full_loss are unnecessary, also i have addedd docsstring.

File: ivy/data_classes/container/experimental/losses.py [hunk 3]
```diff
@@ -160,6 +160,136 @@ def l1_loss(
...
+            Whether to also map method to sequences (lists, tuples).
+            Default is ``False``.
+        out
+            optional output container, for writing the result to. It must have a shape
+            that the inputs broadcast to.
+
+        Returns
+        -------
+        ret
+            The L2 loss between the input array and the targeted values.
```
Comment (KareemMAX) [inline ivy/data_classes/container/experimental/losses.py:206]: I'm a bit confused. What are the targeted values?

File: ivy/functional/ivy/experimental/losses.py [hunk 4]
```diff
@@ -154,6 +154,59 @@ def l1_loss(
...
+    *,
+    reduction: Optional[str] = "sum",
+    out: Optional[ivy.Array] = None,
+) -> ivy.Array:
+    """
+    Computes half the L2 norm without the sqrt:
+
+    Parameters
+    ----------
+    input : Union[ivy.Array, ivy.NativeArray]
```
Comment (KareemMAX) [inline ivy/functional/ivy/experimental/losses.py:174]: [ref](https://unify.ai/docs/ivy/overview/deep_dive/docstrings.html#:~:text=This%20is%20a%20good%20starting%20point.%20But%20we%20need%20to%20make%20some%20changes.%20Firstly%2C%20given%20that%20we%20are%20using%20type%20hints%2C%20repeating%20all%20of%20the%20types%20also%20in%20the%20docs%20would%20be%20a%20needless%20duplication.)
```suggestion
    input
```

File: ivy/functional/ivy/experimental/losses.py [hunk 5]
```diff
@@ -154,6 +154,59 @@ def l1_loss(
...
+        Input array containing input values.
+    reduction : str, optional
+        Reduction method for the output loss. Options:
+        "none" (no reduction), "mean" (mean of losses),
+        "sum" (sum of losses). Default: "mean".
+    out : Optional[ivy.Array], optional
```
Comment (KareemMAX) [inline ivy/functional/ivy/experimental/losses.py:180]: [ref](https://unify.ai/docs/ivy/overview/deep_dive/docstrings.html#:~:text=This%20is%20a%20good%20starting%20point.%20But%20we%20need%20to%20make%20some%20changes.%20Firstly%2C%20given%20that%20we%20are%20using%20type%20hints%2C%20repeating%20all%20of%20the%20types%20also%20in%20the%20docs%20would%20be%20a%20needless%20duplication.)
```suggestion
    out
```

File: ivy/functional/ivy/experimental/losses.py [hunk 6]
```diff
@@ -154,6 +154,59 @@ def l1_loss(
...
+
+    Examples
+    --------
+    >>> x = ivy.array([1,2,3])
+    >>> z = ivy.l2_loss(x)
+    >>> print(z)
+    ivy.array(7.)
+    >>>a = ivy.array([[1,2,3],[4,5,6]])
+    >>>b = ivy.l2_loss(a)
+    >>>print(b)
```
Comment (KareemMAX) [inline ivy/functional/ivy/experimental/losses.py:198]:
```suggestion
    >>> a = ivy.array([[1,2,3],[4,5,6]])
    >>> b = ivy.l2_loss(a)
    >>> print(b)
```

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_nn.py (hunk 2 above)
Comment (KareemMAX) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_nn.py:1406]: Not really sure what you mean. But I think it's safe to remove them

```suggestion
```

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_nn.py [hunk 7]
```diff
@@ -1397,6 +1397,36 @@ def test_tensorflow_log_poisson_loss(
...
+        available_dtypes=helpers.get_dtypes("float"),
+        num_arrays=1,
+    ),
+    compute_full_loss=st.booleans(),
+    test_with_out=st.just(False),
+)
+def test_tensorflow_l2_loss(
+    *,
+    dtype_target_inputs,
+    compute_full_loss,
```
Comment (KareemMAX) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_nn.py:1412]:
```suggestion
```

File: ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_losses.py [hunk 8]
```diff
@@ -97,6 +97,47 @@ def test_l1_loss(
     )
 
 
+@handle_test(
+    fn_tree="functional.ivy.experimental.l2_loss",
+    dtype_input=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes(["float"]),
```
Comment (KareemMAX) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_losses.py:103]: Does this work?
```suggestion
        available_dtypes=helpers.get_dtypes("float"),
```

File: ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_losses.py [hunk 9]
```diff
@@ -97,6 +97,47 @@ def test_l1_loss(
...
+        min_value=1,
+        max_value=100,
```
Comment (KareemMAX) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_losses.py:105]: I feel these should be removed, as they narrows the test space.

```suggestion
```

File: ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_losses.py [hunk 10]
```diff
@@ -97,6 +97,47 @@ def test_l1_loss(
...
+        allow_inf=False,
+    ),
+    dtype_target=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes(["float"]),
+        min_value=1,
+        max_value=100,
+        allow_inf=False,
+    ),
```
Comment (KareemMAX) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_losses.py:113]: What is the target? It not being used in the function

File: ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_losses.py [hunk 11]
```diff
@@ -97,6 +97,47 @@ def test_l1_loss(
...
+    dtype_input, input = dtype_input
+    dtype_target, target = dtype_target
+
+    helpers.test_function(
+        input_dtypes=dtype_input + dtype_target,
+        test_flags=test_flags,
+        backend_to_test=backend_fw,
+        fn_name=fn_name,
+        on_device=on_device,
+        atol_=1e-02,
```
Comment (KareemMAX) [inline ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_losses.py:135]: Can you remove this? Maybe use a safety factor instead

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Pull request closed without merging**